### ISSUE TITLE
Can't use anonymous functions inside arrays

### ISSUE THREAD

**Issue by barisunver** [getdave/wp-intervention#3]
Hi. I can't seem to make this work:

```
$intervention_args = array(
	'resize' => array( 300, 220 ),
	'blur' => 20,
	'text' => array(
		'bla bla',
		50,
		50,
		function( $font ) {
			$font->color( '#FFFFFF' );
		}
	),
	'rotate' => -90
);

$img_src = wp_intervention( $post_image, $intervention_args );
```

Everything works fine except the `function` inside the `array` of the `'text'` argument. It generates the following error:

```
Catchable fatal error: Object of class Closure could not be converted to string in /home/asd/public_html/site/wp-content/plugins/wp-intervention-develop/src/intervention-wrapper.php on line 167
```

When I remove the function, everything including the `'text'` argument works properly. What am I missing here?

**Comment by barisunver** [comment]
Okay, at least I found out that this issue is about naming the cached files. When I replaced this line:

```

```

...with this line:

```

```

...I managed to generate the image with the font settings. I don't really care about the cached image (because I save the image elsewhere afterwards), so my problem's solved but the issue remains.

**Comment by getdave** [comment]
Hi

Thanks v much. Glad to see youre using the plugin.

Someone else on my team pointed this out. Will add hotfix for this soon.

Is there any project online i can go to to see the results of your work? Would be good to see the plugin in action.

All the best

**Comment by barisunver** [comment]
Hi Dave,

Due to a limitation of Intervention Image (can't set boundries for text, so a longer text exceeds the limits of the image while still being single-line) I had to stop using Intervention Image (and naturally wp-intervention) and switch to another GD wrapper called "PHP Image": https://github.com/kus/php-image

Yet that didn't solve my problem either :( I can now use a "textBox" to make a long text multiline, but "centering" that multiline text centers the textBox but not the text within it.

In short, I can't fit a long text as multiline and center it horizontally & vertically.

**Comment by getdave** [comment]
Ok no worries.

I think this pull request fixes the issue youve reported

https://github.com/getdave/wp-intervention/pull/2#issuecomment-286253800
